mempool/iterators.go

```diff
@@ -83,28 +86,28 @@ func (iter *NonBlockingIterator) Next() Entry {
 		panic(fmt.Errorf("Iterator picked a nil entry on lane %d", lane))
 	}
 	iter.cursors[lane] = iter.cursors[lane].Next()
-	iter.counters[lane]++
+	_ = iter.nextLane()
```

Comment:
I am actually also advancing the laneIndex here so that it can start from the next one. This algorithm, unlike the other one, has to iterate through all the lanes for one round. Before we did not need this because the lane index could stay the same between two calls to `PickLane` or `Next` until it hits the count or becomes empty. Here we need to check the next lane on the next call.

Reply:
I see. Could we at least rename `nextLane` to `nextRoundLane` or something meaningful?

Reply:
Yes, I was going to rename the functions now as they are a bit misleading.